test-framework/test_framework.go

```diff
@@ -197,7 +246,37 @@ func newChecker(program *ast.Program) (*sema.Checker, error) {
 					elaboration = stdlib.TestContractChecker.Elaboration
 
 				default:
-					return nil, errors.NewUnexpectedError("importing programs not implemented")
+					importedChecker, err := r.parseAndCheckImport(importedLocation)
+					if err != nil {
+						return nil, err
+					}
+
+					elaboration = importedChecker.Elaboration
+
+					contractDecl := importedChecker.Program.SoleContractDeclaration()
+					compositeType := elaboration.CompositeDeclarationTypes[contractDecl]
+
+					constructorType, constructorArgumentLabels := importedChecker.CompositeConstructorType(contractDecl, compositeType)
+
+					// Remove the contract variable, and instead declare a constructor.
+					elaboration.GlobalValues.Delete(compositeType.Identifier)
+
+					// Declare a constructor
+					_, err = checker.ValueActivations.Declare(sema.VariableDeclaration{
+						Identifier:               contractDecl.Identifier.Identifier,
+						Type:                     constructorType,
+						DocString:                contractDecl.DocString,
+						Access:                   contractDecl.Access,
+						Kind:                     contractDecl.DeclarationKind(),
+						Pos:                      contractDecl.Identifier.Pos,
+						IsConstant:               true,
+						ArgumentLabels:           constructorArgumentLabels,
+						AllowOuterScopeShadowing: false,
+					})
```

Comment:
As far as I understand, the goal here is to declare a contract as a constructor *function*, instead of the default, a *singleton value*?

Instead of having the checker declare the singleton, then removing that declaration, and finally adding a function declaration back, maybe the result of declaring a contract should be configurable in the checker?

That would also avoid having to export all the implementation details (`valueActivations`, etc.).

Reply:
That sounds cleaner. Added a checker option `WithContractVariableHandler`, a one similar to its interpreter counterpart `WithContractValueHandler`: cd433916bc25c2da5cb7322cb6925c10647afd94